### PR TITLE
Enable additional authentication mechanisms in librdkafka

### PR DESCRIPTION
**Pull request by rockwood-openai**
The main reasons from what I can tell for enabling SSL or CURL is to also be able to enabled additional authentication mechanisms

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by rockwood-openai** [comment]
@microsoft-github-policy-service agree company="OpenAI"

**Comment by BillyONeal** [comment]
The title says 'add additional dependencies' but I don't see any dependency changes here. Did you forget to commit `vcpkg.json` changes?

File: versions/l-/librdkafka.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "a84ea0202e606993c8377b34bdbea3c84e08d792",
+      "git-tree": "f707b3b9aa6cade0699134559081a6df068a7643",
```
Comment (BillyONeal) [inline versions/l-/librdkafka.json:4]: Existing versions must not be changed. Please revert changes to this file, set `port-version` in `vcpkg.json` to `1`, and rerun `vcpkg x-add-version`. (There should only be a new version here added to the top, not overwriting an existing version)

**Comment by BillyONeal** [comment]
Please feel free to mark 'ready for review' when you're ready for another review :)

**Comment by rockwood-openai** [comment]
> The title says 'add additional dependencies' but I don't see any dependency changes here. Did you forget to commit `vcpkg.json` changes?

I mean it adds the additional opt-in auth plugins to librdkafka, I've updated the PR title. Thanks!

**Comment by BillyONeal** [comment]
Thanks for the improvements!

**Pull request merged**